### PR TITLE
[WIP] Add aws_guardduty_filter provider

### LINKED ISSUE #9647: AWS GuardDuty: Add support for creating filters
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Terraform aws provider does not currently support the capability to manage filters (CreateFilter, GetFilter, DeleteFilter). 
CloudFormation does support this, so this request is to have parity.

### New or Affected Resource(s)

* GuardDuty
* GuardDuty Filters (example resource name: aws_guardduty_filter)

### Potential Terraform Configuration

We could have something like this:

```hcl
resource "aws_guardduty_filter" "foo" {
   name = "MyFilter"
   description = "Description of MyFilter"
   detector_id = aws_guardduty_detector.id
   rank = 1
   finding_criteria = {
      criterion = {
         updatedAt = {
            Gte = 0
         }
      }
   }
}
```

### References

* https://docs.aws.amazon.com/guardduty/latest/ug/create-filter.html
* https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-guardduty-filter.html

### PR DESCRIPTION
**Pull request by phundisk**
Attempted to add provider per https://github.com/terraform-providers/terraform-provider-aws/issues/9647

Still need to add tests and tags resource but looking for general feedback on my approach and any go best practices since I am new to Go.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Closes #9647 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
Adding provider aws_guardduty_filter resource
Adding tag capability to AWS GuardDuty resources
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: aws/resource_aws_guardduty_filter.go [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+			},
+			"auto_archive": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+			"rank": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"filters": {
```
Comment (gdavison) [inline aws/resource_aws_guardduty_filter.go]: This should be `finding_criteria` to match the API

File: aws/resource_aws_guardduty_filter.go [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
```
Comment (gdavison) [inline aws/resource_aws_guardduty_filter.go]: These elements should match the non-deprecated values of the `guardduty.Condition` struct. You can make all of the attributes `Optional`, and then combinations of `ConflictsWith`, `ExactlyOneOf`, and `AtLeastOneOf` as applicable for the filter combinations.

**Comment by idubinskiy** [comment]
@gdavison Looks like someone also made an attempt in https://github.com/terraform-providers/terraform-provider-aws/pull/10676. Any chance of getting a review for that one, in case it's in a more mergeable state or the opener of it is able to respond more quickly?

**Comment by phundisk** [comment]
Hey I am going to take a look at the comments this weekend and get the PR is a good state!  Thanks so much for your comments @glasser

**Comment by phundisk** [comment]
The one at https://github.com/terraform-providers/terraform-provider-aws/pull/10676/files looks good and a bit more robust than my initial PR.  It might make more sense to get that one merged.  I can take a look this weekend if we decide to go with my initial PR instead

**Comment by gdavison** [comment]
Thanks, @phundisk. I've reviewed #10676

**Comment by gdavison** [comment]
As mentioned above, we're going to go with #10676. I'm going to close this ticket

**Pull request closed without merging**